src/utilities/printSchema.ts

```diff
@@ -36,6 +36,8 @@ import type { GraphQLSchema } from '../type/schema';
 
 import { astFromValue } from './astFromValue';
 
+// TODO: we might need to add a flag to print the schema with semantic-non-null types
```

Comment:
CC @twof @benjie how do we expect to tackle this, document directives aren't allowed as it stands and we'd need to store this on the schema itself. We do have the possibility of a schema directive but that might span different documents.

Reply:
Yeah we'd need to add a new directive scope